pilot/pkg/networking/core/v1alpha3/httproute.go

```diff
@@ -45,6 +46,8 @@ func (configgen *ConfigGeneratorImpl) BuildHTTPRoutes(env *model.Environment, no
 		return configgen.buildSidecarOutboundHTTPRouteConfig(env, node, push, proxyInstances, services, routeName), nil
 	case model.Router, model.Ingress:
 		return configgen.buildGatewayHTTPRouteConfig(env, node, push, proxyInstances, services, routeName)
+	default:
```

Comment:
Is this condition possible?  I find it unlikely that bad input would be passed in...  If there are other NodeTypes that can be decoded, seems like there should be logic to handle them.

Reply:
Thank you for your review.
I also agree it's very unlikely.
However, I think that this method should not know whether it's handled outside.
So I added default handling.